aspnetcore/web-api/http-repl.md

```diff
@@ -0,0 +1,597 @@
+```
+
+The HTTP REPL offers command completion. Pressing the <kbd>Tab</kbd> key iterates through the list of commands that complete the characters or API endpoint that you typed. The following sections outline the available CLI commands. 
+
+## Connect to the web API
+
+Connect to a web API by running the following command:
+
+```console
+dotnet httprepl <BASE URI>
+```
+
+`<BASE URI>` is the base URI for the web API. For example:
+
+```console
+dotnet httprepl https://localhost:5001
+```
+
+Alternatively, run the following command at any time while the HTTP REPL is running:
+
+```console
+set base <BASE URI>
+```
+
+For example:
+
+```console
+(Disconnected)~ set base https://localhost:5001
+```
+
+## Point to the Swagger document for the web API
+
+To properly inspect the web API, set the relative URI to the Swagger document for the web API. To set the relative URI, run the following command:
+
+```console
+set swagger <RELATIVE URI>
+```
+
+For example:
+
+```console
+https://localhost:5001/~ set swagger /swagger/v1/swagger.json
+```
+
+## Navigate the web API
+
+### View available endpoints
+
+To list the different endpoints (controllers) at the root of the web API address, run the `ls` command:
+
+```console
+https://localhot:5001/~ ls
+```
+
+The following output format is displayed:
+
+```console
+.        []
+Fruits   [get|post]
+People   [get|post]
+
+https://localhost:5001/~
+```
+
+The preceding output indicates that there are two controllers available: `Fruits` and `People`. Both controllers support parameterless HTTP GET and POST operations.
+
+Navigating into a specific controller reveals more detail. For example, the following command's output shows the `Fruits` controller also supports HTTP GET, PUT, and DELETE operations. Each of these operations expects an `id` parameter in the route:
+
+```console
+https://localhost:5001/fruits~ ls
+.      [get|post]
+..     []
+{id}   [get|put|delete]
+
+https://localhost:5001/fruits~
+```
+
+Alternatively, run the `ui` command to open the web API's Swagger UI page in a browser. For example:
+
+```console
+https://localhost:5001/~ ui
+```
+
+### Navigate to an endpoint
+
+To navigate to a different endpoint on the web API, run the `cd` command:
+
+```console
+https://localhost:5001/~ cd people
+```
+
+The path following the `cd` command is case insensitive. The following output format is displayed:
+
+```console
+/people    [get|post]
+
+https://localhost:5001/people~
+```
+
+## Customize the HTTP REPL
+
+The HTTP REPL's default [colors](#set-the-default-colors) and [editor](#set-the-default-editor) can be customized. The HTTP REPL preferences are persisted across the current session and are honored in future sessions. Once modified, the preferences are stored in the following file:
+
+# [Linux / macOS](#tab/linux+macos)
+
+*%HOME%\.httpreplprefs*
+
+# [Windows](#tab/windows)
+
+*%USERPROFILE%\.httpreplprefs*
+
+---
+
+### Set the default colors
```

Comment:
Maybe title this "Set color preferences" instead? There are defaults baked in and this is really about overriding them.